diff --git a/src/v4/core/core.ts b/src/v4/core/core.ts
index 2f0facb7ffcc0ee30c916f7f4d30cba00b5c3aee..7deaf0cc67ae553d541eabfa0c19d964b7222ca9 100644
--- a/src/v4/core/core.ts
+++ b/src/v4/core/core.ts
@@ -44,7 +44,7 @@ export /*@__NO_SIDE_EFFECTS__*/ function $constructor<T extends ZodTrait, D = T[
     const keys = Object.keys(proto);
     for (let i = 0; i < keys.length; i++) {
       const k = keys[i]!;
-      if (!(k in inst)) {
+      if (!(k in inst) && typeof proto[k] === "function") {
         (inst as any)[k] = proto[k].bind(inst);
       }
     }
diff --git a/v4/core/core.cjs b/v4/core/core.cjs
index 937f35e921cc9e5db8b717699293fed65a5f0d33..2c88899abc7f2d86bc7ea69efa99b07018e9611e 100644
--- a/v4/core/core.cjs
+++ b/v4/core/core.cjs
@@ -29,7 +29,7 @@ function $constructor(name, initializer, params) {
         const keys = Object.keys(proto);
         for (let i = 0; i < keys.length; i++) {
             const k = keys[i];
-            if (!(k in inst)) {
+            if (!(k in inst) && typeof proto[k] === "function") {
                 inst[k] = proto[k].bind(inst);
             }
         }
diff --git a/v4/core/core.js b/v4/core/core.js
index 2a32024b30237ea02204319642a4619c03ed82d1..8ca1c2c812e77e0483af54019d68762f9e4c2cff 100644
--- a/v4/core/core.js
+++ b/v4/core/core.js
@@ -24,7 +24,7 @@ export /*@__NO_SIDE_EFFECTS__*/ function $constructor(name, initializer, params)
         const keys = Object.keys(proto);
         for (let i = 0; i < keys.length; i++) {
             const k = keys[i];
-            if (!(k in inst)) {
+            if (!(k in inst) && typeof proto[k] === "function") {
                 inst[k] = proto[k].bind(inst);
             }
         }
